Core/GDCore/Events/Builtin/CommentEvent.cpp

```diff
@@ -23,8 +23,9 @@ vector<gd::String> CommentEvent::GetAllSearchableStrings() const {
 
 bool CommentEvent::ReplaceAllSearchableStrings(
     std::vector<gd::String> newSearchableString) {
+  bool isCommentModified = newSearchableString[0] != com1;
   SetComment(newSearchableString[0]);

```

Comment:
If the comment is the same, you could avoid setting the comment and early return false